### PR TITLE
docket: force a |rein %base prior to fixing #5667

### PR DESCRIPTION
**Pull request by yosoyubik**
Tested this is on a ship that had `%hodl` installed before applying https://github.com/urbit/urbit/pull/5732 and all worked fine.

### CODE REVIEW

File: pkg/garden/app/docket.hoon [hunk 1]
```diff
@@ -50,12 +51,15 @@
...
   |^
-  =.  -.state  old
-  =.  +.state  inflate-cache
-  `this
+  =+  !<(old=app-states vase)
+  =?  old    ?=(?(~ ^) -.old)  [%1 old]
+  =^  cards  old
+    ?.  ?=(%1 -.old)  `old
+    :_  old(- %2)
+    [%pass /rein %agent [our.bowl %hood] %poke kiln-rein+!>([%base %.y ~ ~])]~
```
Comment (belisarius222) [inline pkg/garden/app/docket.hoon:60]: won't this run even if we're already at %2?

Comment (belisarius222) [reply pkg/garden/app/docket.hoon:60]: oh, yeah, I can't read good today

**Comment by arguser** [comment]
Logs from failed compilations from (what I think) OTA related to this merge.

```
kiln: %base pausing updates
kiln: %base resuming updates
kiln: resetting tracking for %base from [~ripdys %kids]
kiln: activated install into %base from [~ripdys %kids]
kiln: downloading update for %base from [~ripdys %kids]
kiln: finished downloading update for %base from [~ripdys %kids]
kiln: applying OTA to %base from [~ripdys %kids], kelvin: [lal=%zuse num=419]
kiln: merge into %base succeeded
kiln: merging %base into %kids
>   arvo: beginning upgrade
arvo: compiling next arvo
>   arvo: +load next
lull: ~hidrun-loclun
zuse: ~tipnut-hacpyl
vane %ames: ~rigsud-bolmes
vane %behn: ~sartes-masnyl
vane %clay: ~dachut-hapdur
vane %dill: ~milwer-fogmeb
vane %eyre: ~bonler-ranteb
vane %gall: ~dildus-landef
vane %iris: ~macryl-midnyl
vane %jael: ~rabbyl-famdeg
%clay-kernel-updated
clay: rebuilding %base after kernel update
crud: %drip-fail event failed
~dibref-labter:dojo> drip failed
~dibref-labter:dojo> [%poke %wake]
~dibref-labter:dojo> %load
~dibref-labter:dojo>   bar-stack
~dibref-labter:dojo> ~[
~dibref-labter:dojo>   ~[
~dibref-labter:dojo>     /gall/use/hood/0wLlR6R/~dibref-labter/kiln/vats/base/merge-main
~dibref-labter:dojo>     /dill
~dibref-labter:dojo>     //term/1
~dibref-labter:dojo>   ]
~dibref-labter:dojo>   ~[
~dibref-labter:dojo>     /gall/use/hood/0wLlR6R/~dibref-labter/kiln/vats/base/merge-main
~dibref-labter:dojo>     /dill
~dibref-labter:dojo>     //term/1
~dibref-labter:dojo>   ]
~dibref-labter:dojo>   ~[
~dibref-labter:dojo>     /clay/merge/base/~ripdys/kids/take-that
~dibref-labter:dojo>     /gall/use/hood/0wLlR6R/~dibref-labter/kiln/vats/base/merge-main
~dibref-labter:dojo>     /dill
~dibref-labter:dojo>     //term/1
~dibref-labter:dojo>   ]
~dibref-labter:dojo>   ~[//behn]
~dibref-labter:dojo> ]
~dibref-labter:dojo> call: failed
~dibref-labter:dojo> /sys/vane/clay/hoon:<[4.344 3].[4.585 5]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[4.346 3].[4.585 5]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[4.350 3].[4.585 5]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[4.353 3].[4.585 5]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[4.508 5].[4.513 15]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[4.509 5].[4.513 15]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[4.510 5].[4.513 15]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[4.511 7].[4.512 35]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[4.512 7].[4.512 35]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[4.512 12].[4.512 35]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.535 5].[1.624 20]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.536 5].[1.624 20]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.539 5].[1.624 20]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.546 5].[1.624 20]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.547 5].[1.624 20]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.550 5].[1.624 20]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.554 5].[1.624 20]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.556 5].[1.624 20]>
~dibref-labter:dojo> [ from=17
~dibref-labter:dojo>   deletes={}
~dibref-labter:dojo>     changes
~dibref-labter:dojo>   { [i=~.gen t=/hood/moon-cycle-keys/hoon]
~dibref-labter:dojo>     [i=~.lib t=/azimuth/hoon]
~dibref-labter:dojo>     [i=~.lib t=/hood/helm/hoon]
~dibref-labter:dojo>     [i=~.gen t=/hood/moon/hoon]
~dibref-labter:dojo>     [i=~.app t=/azimuth-tracker/hoon]
~dibref-labter:dojo>     [i=~.app t=/dojo/hoon]
~dibref-labter:dojo>     [i=~.app t=/claz/hoon]
~dibref-labter:dojo>     [i=~.lib t=/pill/hoon]
~dibref-labter:dojo>     [i=~.sys t=/vane/ames/hoon]
~dibref-labter:dojo>     [i=~.ted t=/roller/send/hoon]
~dibref-labter:dojo>     [i=~.gen t=/hood/moon-breach/hoon]
~dibref-labter:dojo>     [i=~.app t=/roller/hoon]
~dibref-labter:dojo>     [i=~.sys t=/arvo/hoon]
~dibref-labter:dojo>     [i=~.app t=/hood/hoon]
~dibref-labter:dojo>     [i=~.tests t=/sys/zuse/crypto/secp256k1/hoon]
~dibref-labter:dojo>     [i=~.lib t=/hood/drum/hoon]
~dibref-labter:dojo>     [i=~.sys t=/lull/hoon]
~dibref-labter:dojo>     [i=~.sys t=/vane/jael/hoon]
~dibref-labter:dojo>     [i=~.app t=/azimuth/hoon]
~dibref-labter:dojo>     [i=~.sur t=/claz/hoon]
~dibref-labter:dojo>     [i=~.app t=/herm/hoon]
~dibref-labter:dojo>     [i=~.lib t=/claz/hoon]
~dibref-labter:dojo>     [i=~.lib t=/shoe/hoon]
~dibref-labter:dojo>     [i=~.sur t=/dice/hoon]
~dibref-labter:dojo>     [i=~.app t=/roller-rpc/hoon]
~dibref-labter:dojo>     [i=~.sys t=/zuse/hoon]
~dibref-labter:dojo>     [i=~.lib t=/dice/hoon]
~dibref-labter:dojo>     [i=~.app t=/eth-watcher/hoon]
~dibref-labter:dojo>     [i=~.lib t=/azimuth-roll-rpc/hoon]
~dibref-labter:dojo>     [i=~.ted t=/eth-watcher/hoon]
~dibref-labter:dojo>   }
~dibref-labter:dojo> ]
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.562 5].[1.624 20]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.563 5].[1.624 20]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.566 5].[1.624 20]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.567 5].[1.624 20]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.568 5].[1.624 20]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.572 5].[1.624 20]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.575 5].[1.624 20]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.576 5].[1.624 20]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.578 5].[1.624 20]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.580 5].[1.624 20]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.580 40].[1.580 71]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.906 7].[1.925 64]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.907 7].[1.925 64]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.911 7].[1.925 64]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.912 7].[1.925 64]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.913 7].[1.925 64]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.915 9].[1.916 49]>
~dibref-labter:dojo> /sys/vane/clay/hoon:<[1.916 9].[1.916 49]>
~dibref-labter:dojo> [%error-validating /app/azimuth/version-0/azimuth-snapshot]
~dibref-labter:dojo> [ %validate-page-fail
~dibref-labter:dojo>   /app/azimuth/version-0/azimuth-snapshot
~dibref-labter:dojo>   %from
~dibref-labter:dojo>   %azimuth-snapshot
~dibref-labter:dojo> ]
~dibref-labter:dojo> [%error-building-dais %azimuth-snapshot]
~dibref-labter:dojo> [%error-building-mark %azimuth-snapshot]
~dibref-labter:dojo> [%error-building /mar/azimuth/snapshot/hoon]
~dibref-labter:dojo> -find.snap-state
~dibref-labter:dojo> /mar/azimuth/snapshot/hoon::[5 10].[5 20]>
~dibref-labter:dojo> /mar/azimuth/snapshot/hoon::[5 5].[5 20]>
~dibref-labter:dojo> /mar/azimuth/snapshot/hoon::[5 1].[20 3]>
```

**Comment by arguser** [comment]
@yosoyubik seems like my sponsor star was offline for some, could be a cause? I see differences on %base and %garden

```
~dibref-labter:dojo> %base
~dibref-labter:dojo>   /sys/kelvin:      [%zuse 419]
~dibref-labter:dojo>   base hash:        ~[0v11.kk4nq.cv61i.s04tr.jajlk.dj407.ohe0p.7pipe.m3n1m.67lu9.9cdv3 0v1f.us55j.qdkqp.7lfdi.7iboh.l6ssv.luq6o.fs8kv.8hgge.q3tqi.t8km6 0veogva.hb5uv.fndh9.01efm.74mvm.6agh0.6js06.783o9.oopop.v938s 0vh.9tiga.vur67.c89ol.3d6rq.mal32.stcqu.s3dds.en8pa.cibvb.11e35]
~dibref-labter:dojo>   %cz hash:         0v1a.u0qbe.7s5n2.a00p4.qg39v.cvjts.beejc.bib88.qnric.7vmtp.hipbp
~dibref-labter:dojo>   app status:       running
~dibref-labter:dojo>   force on:         ~
~dibref-labter:dojo>   force off:        ~
~dibref-labter:dojo>   publishing ship:  ~
~dibref-labter:dojo>   updates:          tracking
~dibref-labter:dojo>   source ship:      ~ripdys
~dibref-labter:dojo>   source desk:      %kids
~dibref-labter:dojo>   source aeon:      16
~dibref-labter:dojo>   pending updates:  ~
~dibref-labter:dojo> ::

```
```
~dibref-labter:dojo> %garden
~dibref-labter:dojo>   /sys/kelvin:      [%zuse 419]
~dibref-labter:dojo>   base hash:        0v18.rb7lh.l7emh.us5ci.moj31.0tbtn.2jl3p.anvkl.qsfgv.5vpk5.bl13m
~dibref-labter:dojo>   %cz hash:         0v1v.6od9s.2048s.39hls.10mom.i55jg.56pc4.e869i.07dmc.mcfmj.0f03b
~dibref-labter:dojo>   app status:       running
~dibref-labter:dojo>   force on:         ~
~dibref-labter:dojo>   force off:        ~
~dibref-labter:dojo>   publishing ship:  [~ ~mister-dister-dozzod-dozzod]
~dibref-labter:dojo>   updates:          tracking
~dibref-labter:dojo>   source ship:      ~mister-dister-dozzod-dozzod
~dibref-labter:dojo>   source desk:      %garden
~dibref-labter:dojo>   source aeon:      19
~dibref-labter:dojo>   pending updates:  ~
~dibref-labter:dojo> ::
```

**Comment by arguser** [comment]
@yosoyubik apparently fixed!

**Pull request merged**